poetry/utils/exporter.py

```diff
@@ -154,6 +154,8 @@ def _export_requirements_txt(
                 url = (
                     repository.authenticated_url if with_credentials else repository.url
                 )
+                if repository.url.startswith("http://"):
```

Comment:
Might be better to use url parsing and checking for parsed `scheme` here. You can see authenticator code to see an example. Besure to use the compat implementation.

Reply:
sure!